src/account/default.ts

```diff
@@ -30,9 +30,10 @@ export class Account extends Provider implements AccountInterface {
 
   private signer: SignerInterface;
 
-  constructor(provider: Provider, address: string, keyPair: KeyPair) {
+  constructor(provider: Provider, address: string, keyPairOrSigner: KeyPair | Signer) {
     super(provider);
-    this.signer = new Signer(keyPair);
+    this.signer =
+      keyPairOrSigner instanceof Signer ? keyPairOrSigner : new Signer(keyPairOrSigner as KeyPair);
```

Comment:
I have bad experiences with `instanceof` not detecting an instance.. Could we change this to something more low level?

Reply:
okay, used "in" typeguard operator in the next commit....